sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/xds/handler/EdsHandler.java

```diff
@@ -0,0 +1,117 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Based on dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/util/protocol/impl/EdsProtocol.java
+ * from the Apache dubbo project.
+ */
+
+package io.sermant.implement.service.xds.handler;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import io.envoyproxy.envoy.config.endpoint.v3.ClusterLoadAssignment;
+import io.envoyproxy.envoy.service.discovery.v3.DiscoveryRequest;
+import io.envoyproxy.envoy.service.discovery.v3.DiscoveryResponse;
+import io.grpc.stub.StreamObserver;
+import io.sermant.core.service.xds.entity.ServiceInstance;
+import io.sermant.core.service.xds.listener.XdsServiceDiscoveryListener;
+import io.sermant.implement.service.xds.cache.XdsDataCache;
+import io.sermant.implement.service.xds.client.XdsClient;
+import io.sermant.implement.service.xds.utils.XdsProtocolTransformer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.logging.Level;
+
+/**
+ * EdsHandler
+ *
+ * @author daizhenyu
+ * @since 2024-05-13
+ **/
+public class EdsHandler extends XdsHandler {
+    /**
+     * construction method
+     *
+     * @param client xds client
+     * @param resourceType xds resource type
+     */
+    public EdsHandler(XdsClient client, String resourceType) {
+        super(client, resourceType);
+    }
+
+    @Override
+    protected void handleResponse(String requestKey, DiscoveryResponse response) {
+        // The contents of the Cds protocol were resolved,
+        // and the mapping between the service name and the cluster was updated
+        Set<ServiceInstance> newInstances = XdsProtocolTransformer
+                .getServiceInstances(decodeResource2ClusterLoadAssignment(response));
+
+        // send ack
+        StreamObserver<DiscoveryRequest> requestObserver = XdsDataCache.REQUEST_OBSERVERS.get(requestKey);
+        requestObserver.onNext(builtAckDiscoveryRequest(response, XdsDataCache.getClustersByServiceName(requestKey)));
+
+        // check whether the service instance has changed
+        if (!isInstanceChanged(XdsDataCache.SERVICE_INSTANCES.get(requestKey), newInstances)) {
+            return;
+        }
+
+        XdsDataCache.SERVICE_INSTANCES.put(requestKey, newInstances);
+
+        // invoke the listener corresponding to service
+        List<XdsServiceDiscoveryListener> listeners = XdsDataCache.SERVICE_DISCOVER_LISTENER
```

Comment:
SERVICE_DISCOVER_LISTENER、REQUEST_OBSERVERS and others in XdsDataCache should be private and provide get method, or it may cause abuse of these data

Reply:
done